src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzer.cs

```diff
@@ -59,7 +59,7 @@ public abstract class SymbolIsBannedAnalyzer<TSyntaxKind> : SymbolIsBannedAnalyz
                 where sourceText != null
                 from line in sourceText.Lines
                 let text = line.ToString()
-                where !string.IsNullOrWhiteSpace(text)
+                where !string.IsNullOrWhiteSpace(text) && text[0] != '#'
```

Comment:
Consider allowing comments at the end of lines too.

Reply:
I thought about that, but wondered if the extra complexity was necessary. Happy to take some feedback.

Reply:
@drewnoakes Constructors are referenced with `.#ctor`. Splitting when `#` is in the middle will break this.

Reply:
`//` could be used for comments instead.

Reply:
Yes, I'm happy to change it to `//`. Should we also include `'` for the VB folks as well?

Reply:
I've made the change so that the comment delimiter is now `//`. Comments can also be included at the end lines.